benchmark/benchmark.cpp

```diff
@@ -0,0 +1,85 @@
+#include <vector>
+#include <iostream>
+#include <utility>
+#include <string>
+#include <cstdlib>
+#include <functional>
+
+#include "utils.h"
+#include "benchmark.h"
+#include "range.h"
+#include "tsc_x86.h"
+#include "../objectives/objectives.h"
+#include "../HGWOSCA/main.h"
+
+
+std::vector<timeInt64> time_algorithm(Config cfg) {
+
+    // First creating function name to function pointer maps
+    auto obj_func_map = create_obj_map();
+    auto algo_func_map = create_algo_map();
+
+    double* mins = filled_array((size_t)cfg.dimension, cfg.min_position);
+    double* maxs = filled_array((size_t)cfg.dimension, cfg.max_position);
+
+    // Bind the parameters such that we have one generic algorithm function to run and benchmark
+    auto algo_func = std::bind(algo_func_map[cfg.algorithm],
+                               obj_func_map[cfg.obj_func],
+                               cfg.population,
+                               cfg.dimension,
+                               cfg.n_iterations,
+                               mins,
+                               maxs);
+
+    std::vector<timeInt64> cycles_vec;
+    double* solution;
+
+    // Run the actual algorithm and time it for n_iterations
+    for (int iter = 0; iter < cfg.n_iterations; ++iter) {
+        timeInt64 start_time = start_tsc();
+
+        solution = algo_func();
+
+        timeInt64 cycles = stop_tsc(start_time);
+
+        cycles_vec.emplace_back(cycles);
+
+        // TODO: Add DEBUG flag to print solutions and timings.
+        // print_solution((size_t)cfg.dimension, solution);
+        // std::cout << " Cycles: " << cycles << std::endl;
+    }
+
+    // TODO: We might want to be able to store all solutions over all runs to see whether they are consistent.
```

Comment:
I would say this should be part of testing, not benchmarking.

Reply:
Makes sense.